electrum/gui/qt/wizard/wallet.py

```diff
@@ -0,0 +1,1382 @@
+        self.wizard_data['multisig_cosigner_data'][str(self.cosigner)] = {}
+
+
+class WCHaveMasterKey(WizardComponent):
+    def __init__(self, parent, wizard):
+        WizardComponent.__init__(self, parent, wizard, title=_('Create keystore from a master key'))
+
+        self.message_create = ' '.join([
+            _("To create a watching-only wallet, please enter your master public key (xpub/ypub/zpub)."),
+            _("To create a spending wallet, please enter a master private key (xprv/yprv/zprv).")
+        ])
+        self.message_cosign = ' '.join([
+            _('Please enter the master public key (xpub) of your cosigner.'),
+            _('Enter their master private key (xprv) if you want to be able to sign for them.')
+        ])
+
+        self.header_layout = QHBoxLayout()
+        self.label = WWLabel()
+        self.label.setMinimumWidth(400)
+        self.header_layout.addWidget(self.label)
+
+        # TODO: KeysLayout assumes too much in parent, refactor KeysLayout
+        # for now, fake parent.next_button.setEnabled
+        class Hack:
+            def setEnabled(self2, b):
+                self.valid = b
+            def setToolTip(self2, b):
+                pass
+        self.next_button = Hack()
+
+    def on_ready(self):
+        # if self.wallet_type == 'standard':
+        #     v = keystore.is_master_key
+        #     self.add_xpub_dialog(title=title, message=message, run_next=self.on_restore_from_key, is_valid=v)
+        # else:
+        #     i = len(self.keystores) + 1
+        #     self.add_cosigner_dialog(index=i, run_next=self.on_restore_from_key, is_valid=keystore.is_bip32_key)
+        if self.wizard_data['wallet_type'] == 'standard':
+            self.label.setText(self.message_create)
+            v = lambda x: bool(keystore.from_master_key(x))
+            self.slayout = KeysLayout(parent=self, header_layout=self.header_layout, is_valid=v,
+                                      allow_multi=False, config=self.wizard.config)
+            self.layout().addLayout(self.slayout)
+        elif self.wizard_data['wallet_type'] == 'multisig':
+            if 'multisig_current_cosigner' in self.wizard_data:
+                self.title = _("Add Cosigner {}").format(self.wizard_data['multisig_current_cosigner'])
+                self.label.setText(self.message_cosign)
+            else:
+                self.wizard_data['multisig_current_cosigner'] = 0
+                self.label.setText(self.message_create)
+            v = lambda x: keystore.is_bip32_key(x)
+            self.slayout = KeysLayout(parent=self, header_layout=self.header_layout, is_valid=v,
+                                      allow_multi=False, config=self.wizard.config)
+            self.layout().addLayout(self.slayout)
+
+    def apply(self):
+        text = self.slayout.get_text()
+        cosigner_data = self.wizard.current_cosigner(self.wizard_data)
+        cosigner_data['master_key'] = text
+
+
+class WCMultisig(WizardComponent):
```

Comment:
Looks like it is not possible to create a wallet that contains multiple `Hardware_KeyStore`s.
If I try to create a 2of2 wallet from a trezor and a ledger, one of the keystores is going to be a software xpub.
This was not the case in the old wizard.

Reply:
fixed in 447a684d1876aaa637c67608ee6f2126e0c70110